include/tcframe/io_manipulator/LineIOSegmentManipulator.hpp

```diff
@@ -29,13 +29,16 @@ class LineIOSegmentManipulator {
 
             if (variable->type() == VariableType::SCALAR) {
                 parseScalar((Scalar*) variable, in);
+                lastVariableName = TokenFormatter::formatVariable(variable->name());
             } else {
                 parseVector((Vector*) variable, size, in);
+                lastVariableName = TokenFormatter::formatVectorElement(variable->name(),
```

Comment:
Refacto `(Vector*) variable` into `vectorVariable` as it is used more than once; also for matrix in the other manipulator